data/src/commonMain/kotlin/com/wire/kalium/logic/data/featureConfig/FeatureConfigModel.kt

```diff
@@ -105,3 +107,38 @@ data class E2EIConfigModel(
     val shouldUseProxy: Boolean,
     val crlProxy: String?,
 )
+
+@Serializable
+sealed interface ChannelFeatureConfiguration {
+
+    @Serializable
+    data object Disabled : ChannelFeatureConfiguration
+
+    @Serializable
+    data class Enabled(
+        /**
+         * Says what user types the team allows to create any kind of channels.
+         */
+        val createChannelsRequirement: TeamUserType,
+
+        /**
+         * Says what user types the team allows to create public channels.
+         */
+        val createPublicChannelsRequirement: TeamUserType,
+    ) : ChannelFeatureConfiguration
```

Comment:
How about Private channels ? users can't just create public channels ?

Reply:
There are two values.

The first one says whether the user can create channels or not (private, public, whatever).

The second one only applies for Public channels. Given that the user can create channels (first value), can it be a public channel?

Reply:
no, I mean do we have a requirement for creating private channels ? 
`        val createPrivateChannelsRequirement: TeamUserType,
`

Reply:
> no, I mean do we have a requirement for creating private channels ? ` val createPrivateChannelsRequirement: TeamUserType,`

There are no requirements for private channels.

Only public channels might have extra requirements

Reply:
The idea is: only show the `Create Channels` button if the user can create channels (createChannelsRequirement).

Only enable `Public` access if the user can create _public_ channels: (createPublicChannelsRequirement)